policy/data/vulnerabilities.go

```diff
@@ -80,3 +65,72 @@ func (ds *DataSource) GetImageVulnerabilities(ctx context.Context, evalCtx goals
 
 	return &query.QueryResponse{}, packages, vulns, nil
 }
+
+// applyVEX returns the CVEs that remain relevant after cross-referencing them with VEX documents.
+func applyVEX(vulnsByPurl types.VulnerabilitiesByPurl, vexDocs []vex.VEX) []types.Vulnerability {
+	filteredOutCVEs := []types.Vulnerability{}
+
+	for _, cve := range vulnsByPurl.Vulnerabilities {
+		for _, v := range vexDocs {
+			for _, stmt := range v.Statements {
+				if cveMatch(cve.SourceId, stmt) {
+					if purlMatch(vulnsByPurl.Purl, stmt) {
+						if notAffectedOrFixed(stmt) {
+							filteredOutCVEs = append(filteredOutCVEs, cve)
+						}
+					}
+				}
+			}
+		}
+	}
+
+	vexedCVEsMap := make(map[string]bool, len(filteredOutCVEs))
+	for _, cve := range filteredOutCVEs {
+		vexedCVEsMap[cve.SourceId] = true
+	}
+
+	// Filter out the VEXed CVEs
+	cves := make([]types.Vulnerability, 0, len(vulnsByPurl.Vulnerabilities))
+	for _, cve := range vulnsByPurl.Vulnerabilities {
+		if !vexedCVEsMap[cve.SourceId] {
+			cves = append(cves, cve)
+		}
+	}
+	return cves
+}
+
+// cveMatch checks whether a CVE is present in a VEX statement
+func cveMatch(cveID string, stmt govex.Statement) bool {
+	return stmt.Vulnerability.ID == cveID || string(stmt.Vulnerability.Name) == cveID
+}
+
+// purlMatch checks whether a purl is present in at least one of the following locations:
+// - Component
+// - Subcomponent(s)
+// - Special case for org-scoped VEXed CVEs.
```

Comment:
Can we have org-scoped VEX statements? All VEX docs come in on a single image, right?

Reply:
*Exceptions* can be org-scoped, but we aren't handling those here.

Reply:
Thanks for clarifying. Isn't the following condition in L125-127 what makes the exception org-scoped?:

```go
if len(p.Subcomponents) == 0 {
 			return true
 		}
 ```

Reply:
No, that's image-scoped, not org-scoped.